lib/glock/conn.ex

```diff
@@ -88,12 +189,51 @@ defmodule Glock.Conn do
     %{conn | connect_opts: merged_opts}
   end
 
-  defp put_opts({:ws_opts, value}, conn) do
+  # If they want no security strip out tls_opts
+  defp put_opts({:transport, :tcp}, %{connect_opts: connect_opts} = conn) do
     %{
       conn
-      | ws_opts: Map.merge(conn.ws_opts, value, fn _key, _default, override -> override end)
+      | connect_opts:
+          Map.delete(connect_opts, :tls_opts) |> Map.update!(:transport, fn _ -> :tcp end)
+    }
+  end
+
+  # If they want to verify the host put that into default tls_opts
+  defp put_opts({:host_name_verify, host_name}, %{connect_opts: connect_opts} = conn) do
+    %{
+      conn
+      | connect_opts: update_in(connect_opts, [:tls_opts], fn _ -> put_verify_peer(host_name) end)
+    }
+  end
```

Comment:
```suggestion
  defp put_opts({:host_name_verify, host_name}, %{connect_opts: connect_opts} = conn) do
    tls_opts = Map.get(connect_opts, :tls_opts)
    %{
      conn
      | connect_opts: update_in(connect_opts, [:tls_opts], fn _ -> [{:verify_fun, {&:ssl_verify_hostname.verify_fun/3, [check_hostname: host_name]}}, {:server_name_indication, host_name}  | Keyword.update!(tls_opts, :verify, fn _ -> :verify_peer end)] end)
    }
  end
```
i'd be more inclined to updating the existing `tls_opts` list rather than replacing it with some swapped values to avoid any unexpected results by folks who might want to customize some of the other tls_opts but combine that with the `:host_name_verify` setting and then have changes overwritten or something. This is sort of a mess, but you get the idea: update the existing `:verify` key and then insert the two others.